### PR TITLE
Add a test for MultiLeRobotDataset making sure it produces all frames.

### PR DESCRIPTION
**Pull request by alexander-soare**
## What this does

As the title suggests.

## How to checkout & try? (for the reviewer)

`DATA_DIR="tests/data" pytest tests/test_datasets.py -vv`

- - -
This change is [Reviewable](https://reviewable.io/reviews/huggingface/lerobot/230)

### CODE REVIEW

File: tests/test_datasets.py [hunk 1]
```diff
@@ -113,6 +114,30 @@ def test_factory(env_name, repo_id, policy_name):
             assert key in item, f"{key}"
 
 
+def test_multilerobotdataset_frames():
+    """Check that all dataset frames are incorporated."""
+    # Note: use the image variants of the dataset to make the test approx 3x faster.
+    repo_ids = ["lerobot/aloha_sim_insertion_human_image", "lerobot/aloha_sim_transfer_cube_human_image"]
+    sub_datasets = [LeRobotDataset(repo_id, root="tests/data") for repo_id in repo_ids]
+    dataset = MultiLeRobotDataset(repo_ids, root="tests/data")
```
Comment (Cadene) [inline tests/test_datasets.py:122]: We dont use root for now. We might when we change the API (DATA_DIR="tests/data" is ugly)
```suggestion
    sub_datasets = [LeRobotDataset(repo_id) for repo_id in repo_ids]
    dataset = MultiLeRobotDataset(repo_ids)
```

File: tests/test_datasets.py [hunk 2]
```diff
@@ -113,6 +114,30 @@ def test_factory(env_name, repo_id, policy_name):
...
+    assert len(dataset) == sum(len(d) for d in sub_datasets)
+    assert dataset.num_samples == sum(d.num_samples for d in sub_datasets)
+    assert dataset.num_episodes == sum(d.num_episodes for d in sub_datasets)
+    # Run through all items of the LeRobotDatasets in parallel with the items of the MultiLerobotDataset and
+    # check they match.
+    expected_dataset_indices = []
+    for i, sub_dataset in enumerate(sub_datasets):
+        expected_dataset_indices.extend([i] * len(sub_dataset))
+    for expected_dataset_index, sub_dataset_item, dataset_item in zip(
```
Comment (Cadene) [inline tests/test_datasets.py:131]: Let it breathe
```suggestion

    for expected_dataset_index, sub_dataset_item, dataset_item in zip(
```

File: tests/test_datasets.py [hunk 3]
```diff
@@ -113,6 +114,30 @@ def test_factory(env_name, repo_id, policy_name):
             assert key in item, f"{key}"
 
 
+def test_multilerobotdataset_frames():
+    """Check that all dataset frames are incorporated."""
+    # Note: use the image variants of the dataset to make the test approx 3x faster.
```
Comment (Cadene) [inline tests/test_datasets.py:119]: Nice

File: tests/test_datasets.py [hunk 4]
```diff
@@ -113,6 +114,30 @@ def test_factory(env_name, repo_id, policy_name):
...
+    repo_ids = ["lerobot/aloha_sim_insertion_human_image", "lerobot/aloha_sim_transfer_cube_human_image"]
+    sub_datasets = [LeRobotDataset(repo_id, root="tests/data") for repo_id in repo_ids]
+    dataset = MultiLeRobotDataset(repo_ids, root="tests/data")
+    assert len(dataset) == sum(len(d) for d in sub_datasets)
+    assert dataset.num_samples == sum(d.num_samples for d in sub_datasets)
+    assert dataset.num_episodes == sum(d.num_episodes for d in sub_datasets)
+    # Run through all items of the LeRobotDatasets in parallel with the items of the MultiLerobotDataset and
```
Comment (Cadene) [inline tests/test_datasets.py:126]: Let it breathe
```suggestion

    # Run through all items of the LeRobotDatasets in parallel with the items of the MultiLerobotDataset and
```

**Review by Cadene — APPROVED** [review]
Thanks. Really cool!

File: tests/test_datasets.py (hunk 2 above)
Comment (alexander-soare) [reply tests/test_datasets.py:131]: :rofl:

**Pull request merged**